packages/jest-snapshot/src/print.ts

```diff
@@ -5,22 +5,54 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import diff, {diffStringsUnified} from 'jest-diff';
+import {
+  DIFF_DELETE,
+  DIFF_EQUAL,
+  DIFF_INSERT,
+  Diff,
+  diffLinesUnified,
+  diffStringsRaw,
+  diffStringsUnified,
+  splitLines0,
+} from 'jest-diff';
 import getType = require('jest-get-type');
 import {
   EXPECTED_COLOR,
+  INVERTED_COLOR,
   RECEIVED_COLOR,
   getLabelPrinter,
-  printDiffOrStringify,
 } from 'jest-matcher-utils';
 import prettyFormat = require('pretty-format');
-import {unescape} from './utils';
+import {unstringifyString} from './utils';
+
+// Given array of diffs, return string:
+// * include common substrings
+// * exclude change substrings which have opposite op
+// * include change substrings which have argument op
+//   with change color only if there is a common substring
+const joinDiffs = (
+  diffs: Array<Diff>,
+  op: number,
+  hasCommon: boolean,
+): string =>
+  diffs.reduce(
+    (reduced: string, diff: Diff): string =>
+      reduced +
+      (diff[0] === DIFF_EQUAL
+        ? diff[1]
+        : diff[0] !== op
+        ? ''
+        : hasCommon
+        ? INVERTED_COLOR(diff[1])
+        : diff[1]),
+    '',
+  );
 
 const isLineDiffable = (received: any): boolean => {
   const receivedType = getType(received);
 
   if (getType.isPrimitive(received)) {
```

Comment:
Simplify to just the string check, no isprimitive?

Reply:
It means for primitive types, only string is line diffable. Would a comment help?

Reply:
Sorry must have misread the code when reviewing on my phone, makes sense to me now.